### PR TITLE
Migrate eloquent integration to sandboxed API

### PR DESCRIPTION
**Pull request by labbati**
### Readiness checklist
- [x] (only for Members) Changelog has been added to the appropriate release draft. Create one if necessary.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the appropriate release draft. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: tests/Integrations/Laravel/V5_7/EloquentTest.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            ]),
+        ], true);
+    }
+
+    public function testUpdate()
+    {
+        $this->connection()->exec("insert into users (email) VALUES ('test-user-updated@email.com')");
+        $traces = $this->tracesFromWebRequest(function () {
+            $spec  = GetSpec::create('Eloquent update', '/eloquent/update');
+            $response = $this->call($spec);
```
Comment (morrisonlevi) [inline tests/Integrations/Laravel/V5_7/EloquentTest.php]: Does `$response` need to get used somehow?

Comment (labbati) [reply tests/Integrations/Laravel/V5_7/EloquentTest.php]: It was just useful to print it out, let me remove from everywhere though, good idea

File: src/DDTrace/Integrations/Eloquent/EloquentSandboxedIntegration.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    /**
+     * Set common values shared by many different spans.
+     *
+     * @param SpanData $span
+     */
+    public function setCommonValues(SpanData $span)
+    {
+        $span->type = Type::SQL;
```
Comment (SammyK) [inline src/DDTrace/Integrations/Eloquent/EloquentSandboxedIntegration.php]: We're missing the service name here. We'll need to add something similar to [this method](https://github.com/DataDog/dd-trace-php/blob/8ca38744500482fe9c2d6e642f227012f49e5fd1/src/DDTrace/Integrations/Laravel/V5/LaravelIntegrationLoader.php#L189-L200) as a static method and then add:

```php
$span->service = EloquentSandboxedIntegration::getAppName();
```

Or grab it in the constructor to reduce the number of method calls and then we could just do something like:

```php
$span->service = $this->serviceName;
```

File: tests/Integrations/Laravel/V4/EloquentTest.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->connection()->exec("DELETE from users where email LIKE 'test-user-%'");
+    }
+
+    protected function expectedServiceName()
+    {
+        // Shouldn't this not be set by teh span encoder?
```
Comment (SammyK) [inline tests/Integrations/Laravel/V4/EloquentTest.php]: The SpanEncoder is only used for userland spans. C-level spans run through [the serializer](https://github.com/DataDog/dd-trace-php/blob/8ca38744500482fe9c2d6e642f227012f49e5fd1/src/ext/serializer.c#L314-L342). Once we get the service name worked out (from the earlier comment) the service names should be the same for sandboxed and non-sandboxed integrations. :)

File: tests/Integrations/Laravel/V5_7/EloquentTest.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+namespace DDTrace\Tests\Integrations\Laravel\V5_7;
+
+use DDTrace\Tests\Common\SpanAssertion;
+use DDTrace\Tests\Common\SpanAssertionTrait;
+use DDTrace\Tests\Common\TracerTestTrait;
+use DDTrace\Tests\Common\WebFrameworkTestCase;
+use DDTrace\Tests\Frameworks\Util\Request\GetSpec;
+
+class EloquentTest extends WebFrameworkTestCase
```
Comment (SammyK) [inline tests/Integrations/Laravel/V5_7/EloquentTest.php]: Awesome job on setting up these extra web endpoints to test! 👍

File: src/DDTrace/Integrations/Eloquent/EloquentSandboxedIntegration.php (hunk 2 above)
Comment (labbati) [reply src/DDTrace/Integrations/Eloquent/EloquentSandboxedIntegration.php]: Great catch @SammyK

**Pull request merged**